pkg/daemon/rpm-ostree.go

```diff
@@ -77,8 +79,14 @@ func NewNodeUpdaterClient() NodeUpdaterClient {
 	return &RpmOstreeClient{}
 }
 
-// GetBootedDeployment returns the current deployment found
-func (r *RpmOstreeClient) GetBootedDeployment() (*RpmOstreeDeployment, error) {
+// Synchronously invoke rpm-ostree, writing its stdout to our stdout,
+// and gathering stderr into a buffer which will be returned in err
+// in case of error.
+func runRpmOstree(args ...string) error {
```

Comment:
I think we will also need to update Rebase() to call runRpmOstree()

Reply:
Good observation!  Thinking about this a bit though, the key thing in this change is that we're invoking the `Initialize()` method early on.  `runRpmOstree` is just a wrapper that uses the cleaned up subprocess handling in `runCmdSync` - but it doesn't actually have any of the new special rpm-ostree logic itself.

So while I think you're correct in that this change better matches what lives on master with that suggestion, I don't think it's actually *required* either.

But regardless - updated!